packages/runtime/container-runtime/src/runningSummarizer.ts

```diff
@@ -121,15 +120,22 @@ export class RunningSummarizer implements IDisposable {
             },
         );
 
-        if (!disableHeuristics) {
+        if (configuration.state !== "disableHeuristics") {
+            assert(this.configuration.state === "enabled", "Configuration state should be enabled");
+            const heuristics: ISummaryConfigurationHeuristicSettings = {
+                    ...this.configuration,
```

Comment:
Why is this copy required? Might want to add comment if it is, as it's not clear

Reply:
I will add a comment.